servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcExceptionMapperServiceFilter.java

```diff
@@ -0,0 +1,141 @@
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.api;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.concurrent.api.Single;
+import io.servicetalk.http.api.HttpExecutionStrategies;
+import io.servicetalk.http.api.HttpExecutionStrategy;
+import io.servicetalk.http.api.HttpHeaders;
+import io.servicetalk.http.api.HttpServiceContext;
+import io.servicetalk.http.api.StatelessTrailersTransformer;
+import io.servicetalk.http.api.StreamingHttpRequest;
+import io.servicetalk.http.api.StreamingHttpResponse;
+import io.servicetalk.http.api.StreamingHttpResponseFactory;
+import io.servicetalk.http.api.StreamingHttpService;
+import io.servicetalk.http.api.StreamingHttpServiceFilter;
+import io.servicetalk.http.api.StreamingHttpServiceFilterFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.servicetalk.concurrent.api.Single.succeeded;
+import static io.servicetalk.grpc.api.GrpcHeaderNames.GRPC_STATUS;
+import static io.servicetalk.grpc.api.GrpcHeaderValues.APPLICATION_GRPC;
+import static io.servicetalk.grpc.api.GrpcStatusCode.OK;
+import static io.servicetalk.grpc.api.GrpcUtils.STATUS_OK;
+import static io.servicetalk.grpc.api.GrpcUtils.newErrorResponse;
+import static io.servicetalk.grpc.api.GrpcUtils.setStatus;
+
+/**
+ * Filter that maps known {@link Throwable} subtypes into an gRPC response with an appropriate {@link GrpcStatusCode}.
+ * <p>
+ * This filter is recommended to be placed as early as possible to make sure it captures all exceptions that may be
+ * generated by other filters.
+ */
+public final class GrpcExceptionMapperServiceFilter implements StreamingHttpServiceFilterFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GrpcExceptionMapperServiceFilter.class);
+
+    /**
+     * Instance of {@link GrpcExceptionMapperServiceFilter}.
+     */
+    public static final StreamingHttpServiceFilterFactory INSTANCE = new GrpcExceptionMapperServiceFilter();
+
+    private GrpcExceptionMapperServiceFilter() {
+        // Singleton
+    }
+
+    @Override
+    public StreamingHttpServiceFilter create(final StreamingHttpService service) {
+        return new StreamingHttpServiceFilter(service) {
+
+            @Override
+            public Single<StreamingHttpResponse> handle(final HttpServiceContext ctx,
+                                                        final StreamingHttpRequest request,
+                                                        final StreamingHttpResponseFactory responseFactory) {
+                final BufferAllocator allocator = ctx.executionContext().bufferAllocator();
+                final Single<StreamingHttpResponse> handle;
+                try {
+                    handle = delegate().handle(ctx, request, responseFactory).map(response -> {
+                        // FIXME: make this logic optional for users who want to be able to respond for HTTP requests
```

Comment:
Actually, this change improves one thing (missed `grpc-status` if a filter throws) but breaks another (users won't be able to respond to HTTP requests in a filter). I will defer integrating this PR until we have a way to modify filters chain.